src/libs/xapi-stats.ts

```diff
@@ -0,0 +1,438 @@
+import defaults from 'lodash/defaults.js'
+import findKey from 'lodash/findKey.js'
+import forEach from 'lodash/forEach.js'
+import identity from 'lodash/identity.js'
+import map from 'lodash/map.js'
+import { BaseError } from 'make-error'
+import type XenApi from '@/libs/xen-api'
+import { useHostStore } from '@/stores/host.store'
+import type { XenApiHost } from '@/libs/xen-api'
+import { limitConcurrency } from 'limit-concurrency-decorator'
+import { synchronized } from 'decorator-synchronized'
+import { useVmStore } from '@/stores/vm.store'
+
+class FaultyGranularity extends BaseError {}
+
+// -------------------------------------------------------------------
+
+// according to https://xapi-project.github.io/xen-api/metrics.html
+// The values are stored at intervals of:
+//  - 5 seconds for the past 10 minutes
+//  - one minute for the past 2 hours
+//  - one hour for the past week
+//  - one day for the past year
+const RRD_STEP_SECONDS = 5
+const RRD_STEP_MINUTES = 60
+const RRD_STEP_HOURS = 3600
+const RRD_STEP_DAYS = 86400
+
+export enum Granularity {
+  Seconds = "seconds",
+  Minutes = "minutes",
+  Hours = 'hours',
+  Days = 'days'
+}
+
+const RRD_STEP_FROM_STRING: {[key:string]: number} = {
+  seconds: RRD_STEP_SECONDS,
+  minutes: RRD_STEP_MINUTES,
+  hours: RRD_STEP_HOURS,
+  days: RRD_STEP_DAYS,
+}
+
+// points = intervalInSeconds / step
+const RRD_POINTS_PER_STEP :{[key: string]: number} ={
+  [RRD_STEP_SECONDS]: 120,
+  [RRD_STEP_MINUTES]: 120,
+  [RRD_STEP_HOURS]: 168,
+  [RRD_STEP_DAYS]: 366,
+}
```

Comment:
Petit proposition de typage (il faudra peut-être modifier le reste du code pour adapter, évidemment).

Sinon, ce n'est pas très parlant RRD. Ça veut dire quoi ? Peut-on envisager un nom plus explicite ?

```suggestion
enum RRD_STEP {
  Seconds = 5,
  Minutes = 60,
  Hours= 3600,
  Days = 86400,
}

export enum GRANULARITY {
  Seconds = "seconds",
  Minutes = "minutes",
  Hours = "hours",
  Days = "days",
}

const RRD_STEP_FROM_STRING: {[key in GRANULARITY]: RRD_STEP} = {
  [GRANULARITY.Seconds]: RRD_STEP.Seconds,
  [GRANULARITY.Minutes]: RRD_STEP.Minutes,
  [GRANULARITY.Hours]: RRD_STEP.Hours,
  [GRANULARITY.Days]: RRD_STEP.Days,
}

// points = intervalInSeconds / step
const RRD_POINTS_PER_STEP :{[key in RRD_STEP]: number} = {
  [Steps.Seconds]: 120,
  [Steps.Minutes]: 120,
  [Steps.Hours]: 168,
  [Steps.Days]: 366,
}
```

Reply:
RRD means Round Robin  Database